### PR TITLE
parse nef file scripts

### PR DESCRIPTION
**Pull request by Hecate2**
# Description

Parsing base64(.nef file) directly

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update (Maybe not needed. No significant change from users' perspective)

# How Has This Been Tested?

- [x] Test A
`parse TkVGM1Rlc3RpbmdFbmdpbmUAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAP3LAVcAAnl4ukBXAAJ5eLlAVwABeJlAVwABeJpAVwACeXigSgMAAAAAAAAAgAQAAAAAAAAAgAAAAAAAAAAAuyQDOkBXAAJ5eEpRTqFKUaBRUJ9KEAEAAbskAzoSwEBXAAJ5eEpRTqFKUaBRUJ9KAQCAAgCAAAC7JAM6EsBAVwACeXhKUU6hSlGgUVCfSgIAAACAAwAAAIAAAAAAuyQDOhLAQFcAAnl4SlFOoUpRoFFQn0oDAAAAAAAAAIAEAAAAAAAAAIAAAAAAAAAAALskAzoSwEBXAAJ5eEpRTqFKUaBRUJ9KAIABgAC7JAM6EsBAVwACeXhKUU6hSlGgUVCfShACAAABALskAzoSwEBXAAJ5eEpRTqFKUaBRUJ9KEAMAAAAAAQAAALskAzoSwEBXAAJ5eEpRTqFKUaBRUJ9KEAQAAAAAAAAAAAEAAAAAAAAAuyQDOhLAQFcAA3h5ektLMgM6U7q5QFcAA3h5ektLMgM6U7q5QFcAA3h5ektLMgM6U7q5QFcAA3h5ektLMgM6U7q5QFcAA3h5ektLMgM6U7q5QFcAA3h5ektLMgM6U7q5QFcAA3h5ektLMgM6U7q5QFcAA3h5ektLMgM6U7q5QNYWGhM=`

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by shargon — APPROVED** [review]

**Review by superboyiii — APPROVED** [review]

**Comment by superboyiii** [comment]
Maybe we can also support NEF file to Base64 ouput, it's useful in some rpc cases.

**Comment by Hecate2** [comment]
> Maybe we can also support NEF file to Base64 ouput, it's useful in some rpc cases.

Do we input a path to the NEF file for that? I guess the `parse` feature is used at https://neo.org/converter/index , and the input of a path may affect that service.

**Comment by superboyiii** [comment]
> Do we input a path
Yes, we can input a path and output nef base64 data

File: src/Neo.CLI/CLI/MainService.Tools.cs [hunk 1]
```diff
@@ -441,6 +442,53 @@ private static string Base64Fixed(string str)
             }
         }
 
+        /// <summary>
+        /// Base64 .nef file Analysis
+        /// </summary>
+        [ParseFunction("Base64 .nef file Analysis")]
+        private string? NefFileAnalyis(string base64)
```
Comment (AnnaShaleva) [inline src/Neo.CLI/CLI/MainService.Tools.cs:449]: While we're here, I'd suggest to support hex and (if possible) filename as an input. E.g. in NeoGo we have similar commands that handle both base64- and hex- encoded NEF  and also it's able to read NEF from file using provided filepath, ref. https://github.com/nspcc-dev/neo-go/blob/6bbf15f30586b927d3a1e398688107fb0f38daf1/cli/vm/cli.go#L763, https://github.com/nspcc-dev/neo-go/blob/6bbf15f30586b927d3a1e398688107fb0f38daf1/cli/vm/cli.go#L803 and https://github.com/nspcc-dev/neo-go/blob/6bbf15f30586b927d3a1e398688107fb0f38daf1/cli/vm/cli.go#L692.

**Review by AnnaShaleva — COMMENTED** [review]
Quite a useful extension.

**Review by Jim8y — DISMISSED** [review]

**Review by vncoelho — DISMISSED** [review]

**Review by AnnaShaleva — APPROVED** [review]

**Comment by AnnaShaleva** [comment]
Format job is failing.

**Comment by Hecate2** [comment]
> Format job is failing.

Formatting is OK on my computer. I think a rerun of action is enough.

**Review by shargon — APPROVED** [review]

**Review by Jim8y — APPROVED** [review]

**Pull request merged**